src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,340 @@
+            for (List<ExternalObjectDirectoryEntity> eodsForBatch : batchesForArm) {
+                log.info("Creating DETS batch {} out of {}", batchCounter++, batchesForArm.size());
+                createAndSendBatchFile(eodsForBatch, userAccount);
+            }
+        }
+        log.info("Finished running DETS ARM Batch Push processing at: {}", OffsetDateTime.now());
+    }
+
+    private List<ExternalObjectDirectoryEntity> getDetsEodEntitiesToSendToArm(ExternalLocationTypeEntity sourceLocation,
+                                                                              ExternalLocationTypeEntity armLocation, int maxResultSize) {
+        ObjectRecordStatusEntity armRawStatusFailed = EodHelper.failedArmRawDataStatus();
+        ObjectRecordStatusEntity armManifestFailed = EodHelper.failedArmManifestFileStatus();
+
+        List<ObjectRecordStatusEntity> failedArmStatuses = List.of(armRawStatusFailed, armManifestFailed);
+
+        var failedArmExternalObjectDirectoryEntities = externalObjectDirectoryRepository.findNotFinishedAndNotExceededRetryInStorageLocationForDets(
+            failedArmStatuses,
+            armLocation,
+            armDataManagementConfiguration.getMaxRetryAttempts(),
+            Pageable.ofSize(maxResultSize)
+        );
+
+        List<ExternalObjectDirectoryEntity> returnList = new ArrayList<>(failedArmExternalObjectDirectoryEntities);
+
+        int remainingBatchSizeEods = maxResultSize - failedArmExternalObjectDirectoryEntities.size();
+        if (remainingBatchSizeEods > 0) {
+            var pendingUnstructuredExternalObjectDirectoryEntities = externalObjectDirectoryRepository.findEodsNotInOtherStorage(
+                EodHelper.storedStatus(), sourceLocation,
+                EodHelper.armLocation(),
+                remainingBatchSizeEods);
+            returnList.addAll(pendingUnstructuredExternalObjectDirectoryEntities);
+        }
+
+        return returnList;
+    }
+
+    private void createAndSendBatchFile(List<ExternalObjectDirectoryEntity> eodsForBatch, UserAccountEntity userAccount) {
+        File archiveRecordsFile = dataStoreToArmHelper.createEmptyArchiveRecordsFile(getManifestFilePrefix());
+        var batchItems = new ArmBatchItems();
+
+        for (var currentEod : eodsForBatch) {
+            ObjectStateRecordEntity objectStateRecord = null;
+            var batchItem = new ArmBatchItem();
+            try {
+                ExternalObjectDirectoryEntity armEod;
+                if (isEqual(currentEod.getExternalLocationType(), EodHelper.armLocation())) {
+                    //retry existing attempt that has previously failed.
+                    armEod = currentEod;
+                    batchItem.setArmEod(armEod);
+                    dataStoreToArmHelper.updateArmEodToArmIngestionStatus(
+                        currentEod, batchItem, batchItems, archiveRecordsFile, userAccount, EodHelper.detsLocation());
+                    getObjectStateRecordEntity(armEod);
```

Comment:
was this meant to be:
```
objectStateRecord =  getObjectStateRecordEntity(armEod);
```
?

Reply:
We aren't using the object state record here but we are verifying that it exists. If not an exception is thrown and this EOD will be skipped.